go/plugins/googleai/googleai.go

```diff
@@ -36,7 +58,7 @@ type Config struct {
 	APIKey string
 	// Generative models to provide.
 	// If empty, a complete list will be obtained from the service.
-	Models []string
+	Models map[string]*ai.ModelCapabilities
```

Comment:
Just for new models that haven't been released by us yet.

Reply:
Alex, the idea here is that developers who are just starting out would leave `Models` blank, and the plugin would call the `ListModels` API and fetch all available models. It's good for developers who are just starting out, but the idea is that once they've built everything they would provide the final list of model they are using so that the framework doesn't have to do `ListModels` on cold start. 

Reading that back, now I'm convinced that `ListModels` is not ideal. The bottom line is that we need `ai.ModelCapabilities` one way or the other. Either the user must provide them (which is bad DX and error prone) or we have to hardcode them.  The approach we took on JS side is the latter -- all known models are hardcoded in the plugin with their capabilities. If a new model comes out that we're not aware of only then the developer will have to manually specify `ai.ModelCapabilities` (or wait until we add the model to the plugin).

Reply:
I like/agree with the last takeaway. Hardcoding what we know and allowing some way to use newly released models with manual settings feels like the right way to go since new models don't drop every week.

I didn't know about the `ListModels` API.